### PR TITLE
[macOS Ventura] Disable code signing in TestWebKitAPI and WebKitTestRunner

### PR DESCRIPTION
**Pull request by emw-apple**
#### 4febba7c264b9476b65518b1f492ca5c7726b73c
<pre>
Fix xcconfig and reland &quot;[macOS Ventura] Disable code signing in TestWebKitAPI and WebKitTestRunner&quot;
<a href="https://bugs.webkit.org/show_bug.cgi?id=242730">https://bugs.webkit.org/show_bug.cgi?id=242730</a>

Reviewed by NOBODY (OOPS!).

Variable expansions were missing a layer of parentheses.

* Tools/TestWebKitAPI/Configurations/TestWebKitAPI.xcconfig:
* Tools/WebKitTestRunner/Configurations/WebKitTestRunner.xcconfig:
</pre>

### CODE REVIEW

File: Tools/WebKitTestRunner/Configurations/WebKitTestRunner.xcconfig [hunk 1]
```diff
@@ -45,12 +45,5 @@ INCLUDED_SOURCE_FILE_NAMES = mac/main.mm;
 
 CODE_SIGN_ENTITLEMENTS[sdk=macosx*] = Configurations/WebKitTestRunner.entitlements;
 
-CODE_SIGN_IDENTITY[sdk=macosx*] = $(CODE_SIGN_IDENTITY_$(CONFIGURATION));
```
Comment (aproskuryakov) [inline Tools/WebKitTestRunner/Configurations/WebKitTestRunner.xcconfig:48]: I'm slightly curious about why TestWebKitAPI needs CODE_SIGN_IDENTITY in two xcconfigs, but WebKitTestRunner only needs it in DebugRelease.

**Review by aproskuryakov — APPROVED** [review]
I think that "-" is called ad hoc, not null. Also, I'm not sure if there is any "default" identity, so the commit message seems a little misleading.

**Comment by aj062** [comment]
Cancelled api-mac build https://ews-build.webkit.org/#/builders/3/builds/76641 as it was failing and stuck in retry loop (e.g.: https://ews-build.webkit.org/#/builders/3/builds/76633)

File: Tools/WebKitTestRunner/Configurations/WebKitTestRunner.xcconfig (hunk 1 above)
Comment (emw-apple) [reply Tools/WebKitTestRunner/Configurations/WebKitTestRunner.xcconfig:48]: WebKitTestRunner has these same CODE_SIGN_IDENTITY declarations already: https://github.com/WebKit/WebKit/blob/cb8e510d4f433626e5fe6f9ab6333e46f8b337b6/Tools/WebKitTestRunner/Configurations/DebugRelease.xcconfig#L33-L36

So this change aligns TestWebKitAPI's signing identity behavior with WKTR's.

Comment (emw-apple) [reply Tools/WebKitTestRunner/Configurations/WebKitTestRunner.xcconfig:48]: Actually, I was misreading, you were right to point out the discontinuity.

Updated both projects to set ad-hoc codesigning unconditionally outside of DebugRelease. This is other production-buildable tools (DumpRenderTree, MiniBrowser, MobileMiniBrowser) do.

**Comment by emw-apple** [comment]
Infra issue while running tests, but no build breakage. I've also verified that this doesn't disrupt signing for internal engineering and production builds.

**Comment by emw-apple** [comment]
Switching these tools to ad-hoc code signing broke EWS, and their respective test workflows were crashing:
```
Process:               WebKitTestRunner [8854]
…
Exception Type:        EXC_CRASH (Code Signature Invalid)
```

As far as I can tell, these tools are _never_ signed by build.webkit.org, so the proper approach to take here is to set `CODE_SIGNING_ALLOWED=NO` and prevent Xcode from trying to sign using an empty identity.

**Review by JonWBedard — APPROVED** [review]
r+, although we really should wait to make sure EWS sim can run tests before landing this.

**Comment by emw-apple** [comment]
Confirmed that EWS sims can run tests, and that Apple Silicon machines can still build and run tests.

**Pull request merged**